src/main/java/hudson/remoting/URLish.java

```diff
@@ -16,20 +18,28 @@
 abstract class URLish {
     private URLish() {}
 
+    /**
+     * Converts URLish to the standard {@link URL} type.
+     * @return URL or {@code null} if the target destination is known to be non-existent.
```

Comment:
We could, of course.
Can do it though now we are rather in the code refactoring area than in the NPE warning fix one